src/main/scala/dpla/ingestion3/ReporterMain.scala

```diff
@@ -0,0 +1,95 @@
+package dpla.ingestion3
+
+import org.apache.log4j.{Logger, LogManager}
+import dpla.ingestion3.reports._
+import util.{Success, Failure}
+
+
+/*
+ * ReporterMain and Reporter, for generating QA reports
+ *
+ * Example invocation:
+ *
+  *   $ sbt "run-main dpla.ingestion3.ReporterMain \
+  *     /path/to/enriched-data.avro /path/to/propvalreport local[2] \
+  *     propertyValue sourceResource.type"
+ */
+
+
+/**
+  * Reporter, the report driver class.
+  *
+  * This is a class that is instantiated by ReporterMain because the original
+  * idea was it would be possible to unit test this way.  Note that getReport()
+  * is probably the only method that is worth testing.  See the unit tests in
+  * ReporterTest.
+  *
+  * @see ReporterTest
+  *
+  * @param token            The report name token
+  * @param inputURI         Input URI or file path
+  * @param outputURI        Output URI or file path
+  * @param sparkMasterName  Spark master name, e.g. "local[1]"
+  * @param reportParams     Additional parameters particular to the report
+  */
+class Reporter (
+                 token: String,
+                 inputURI: String,
+                 outputURI: String,
+                 sparkMasterName: String,
+                 reportParams: Array[String]
+               ) {
+
+  val logger: Logger = LogManager.getLogger(this.getClass)
+
+  private def getReport(token: String): Option[Report] = {
+    token match {
+      case "propertyValue" =>
+        Some(new PropertyValueReport(
+          inputURI, outputURI, sparkMasterName, reportParams
+        ))
+      case _ => None
+    }
+  }
+
+  def main(): Unit = {
+    val report = getReport(token)
+    if (report.isDefined) {
```

Comment:
Yes, that's better, let me try that. Codacy complains about the `report.get()`, too